### PR TITLE
PHPLIB-1153: Add `FunctionalTestCase::skipIfServerVersion($op, $version)`

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPLIB-1153

### CODE REVIEW

File: tests/FunctionalTestCase.php [hunk 1]
```diff
@@ -484,6 +484,13 @@ protected function isShardedClusterUsingReplicasets()
         return preg_match('@^.*/.*:\d+@', $document['host']);
     }
 
+    protected function skipIfServerVersionNotSupported(string $operator, string $version, ?string $message = null): void
```
Comment (jmikola) [inline tests/FunctionalTestCase.php:487]: How do you feel about naming this `skipIfServerVersion()? That would both make the LOC shorter and allow it to read like a natural sentence. For example:

```php
$this->skipIfServerVersion('<', '7.0.0', '...');
```

File: tests/Operation/CreateIndexesFunctionalTest.php [hunk 2]
```diff
@@ -216,9 +213,7 @@ function (array $event): void {
 
     public function testCommitQuorumUnsupported(): void
     {
-        if (version_compare($this->getServerVersion(), '4.3.4', '>=')) {
-            $this->markTestSkipped('commitQuorum is supported');
-        }
+        $this->skipIfServerVersionNotSupported('>=', '4.3.4', 'commitQuorum is supported');
```
Comment (jmikola) [inline tests/Operation/CreateIndexesFunctionalTest.php:216]: Note: this is one example where "NotSupported" in the method name doesn't actually make sense. We're intentionally skipping a test where a feature _is_ supported in the server.

Likewise for some `hint` tests below.

File: tests/Operation/WatchFunctionalTest.php [hunk 3]
```diff
@@ -61,9 +60,7 @@ public function setUp(): void
      */
     public function testGetResumeToken(): void
     {
-        if ($this->isPostBatchResumeTokenSupported()) {
-            $this->markTestSkipped('postBatchResumeToken is supported');
-        }
+        $this->skipIfPostBatchResumeTokenNotSupported();
```
Comment (jmikola) [inline tests/Operation/WatchFunctionalTest.php:63]: This logic is incorrect. The test needs to be skipped if `postBatchResumeToken` _is_ supported (i.e. server version >= 4.0.7).

File: tests/Operation/WatchFunctionalTest.php [hunk 4]
```diff
@@ -1596,9 +1567,9 @@ private function insertDocument($document): void
         $this->assertEquals(1, $writeResult->getInsertedCount());
     }
 
-    private function isPostBatchResumeTokenSupported()
+    private function skipIfPostBatchResumeTokenNotSupported(): void
     {
-        return version_compare($this->getServerVersion(), '4.0.7', '>=');
+        $this->skipIfServerVersionNotSupported('>=', '4.0.7', 'postBatchResumeToken is supported');
```
Comment (jmikola) [inline tests/Operation/WatchFunctionalTest.php:1572]: As noted above, this method doesn't cover all use cases for `isPostBatchResumeTokenSupported()`. Three tests are skipped when `postBatchResumeToken` _is_ supported, and two are skipped when it's _not_ supported.

If you'd like to refactor this, I'd propose just dropping this method entirely and replacing the calling code with the appropriate `$this->skipIfServerVersion()` call. It's unlikely that we'd add many more tests for the feature so I'm not concerned with duplicating "4.0.7" in a few places.

**Review by jmikola — CHANGES_REQUESTED** [review]

File: tests/FunctionalTestCase.php (hunk 1 above)
Comment (GromNaN) [reply tests/FunctionalTestCase.php]: Less words is more correct 🥂

File: tests/Operation/WatchFunctionalTest.php (hunk 4 above)
Comment (GromNaN) [reply tests/Operation/WatchFunctionalTest.php]: Good idea, I've totally removed `isPostBatchResumeTokenSupported`. This has more sense.

**Review by jmikola — APPROVED** [review]
One suggestion for the assertion message.

LGTM after that is fixed.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**